crates/core/src/metadata_store.rs

```diff
@@ -299,28 +306,43 @@ impl MetadataStoreClient {
         &self,
         key: ByteString,
     ) -> Result<Option<T>, ReadError> {
-        let value = self.inner.get(key).await?;
+        let start_time = Instant::now();
+        counter!(METADATA_CLIENT_GET_TOTAL, "key" => key.to_string()).increment(1);
```

Comment:
I'm a little worried about the cardinality of this counter. We have a key per partition for epoch metadata.

Reply:
Yeah, i wasn't sure about this tbh. On other hand it also can help debugging irregular access pattern to a certain key.

What about if the key is prefixed with `pp_epoch` we only use that prefix for the metrics?

Reply:
it's okay, let's leave the key for counters, but histogram per key is not great.